MetadataExtractor/IO/IndexedReader.cs

```diff
@@ -318,18 +318,39 @@ public string GetString(int index, int bytesRequested, [NotNull] Encoding encodi
         /// The maximum number of bytes to read.  If a zero-byte is not reached within this limit,
         /// reading will stop and the string will be truncated to this length.
         /// </param>
-        /// <returns>The read string.</returns>
+        /// <returns>The read <see cref="string"/></returns>
         /// <exception cref="System.IO.IOException">The buffer does not contain enough bytes to satisfy this request.</exception>
         [NotNull]
         public string GetNullTerminatedString(int index, int maxLengthBytes)
         {
+            return GetNullTerminatedStringValue(index, maxLengthBytes).ToString();
+        }
+
+        /// <summary>
+        /// Creates a string starting at the specified index, and ending where either <c>byte=='\0'</c> or
+        /// <c>length==maxLength</c>.
+        /// </summary>
+        /// <param name="index">The index within the buffer at which to start reading the string.</param>
+        /// <param name="maxLengthBytes">
+        /// The maximum number of bytes to read.  If a zero-byte is not reached within this limit,
+        /// reading will stop and the string will be truncated to this length.
+        /// </param>
+        /// <returns>The read <see cref="StringValue"/></returns>
+        /// <exception cref="System.IO.IOException">The buffer does not contain enough bytes to satisfy this request.</exception>
+        public StringValue GetNullTerminatedStringValue(int index, int maxLengthBytes)
+        {
             // NOTE currently only really suited to single-byte character strings
```

Comment:
Agreed, since it's now just grabbing bytes. The previous code might have just had the comment out of place. The call `return Encoding.UTF8.GetString(bytes, 0, length);` in the old code might have been the intended line. Whether the comment even described that issue correctly, I'm not sure but it doesn't seem like it needs to be there anymore.